libp2p/connection.nim

```diff
@@ -44,11 +44,16 @@ proc init*[T: Connection](self: var T, stream: LPStream) =
   # bind stream's close event to connection's close
   # to ensure correct close propagation
   let this = self
-  if not isNil(self.stream.closeEvent):
-    self.stream.closeEvent.wait().
+  if not isNil(stream.closeEvent):
+    stream.closeEvent.wait().
       addCallback do (udata: pointer):
```

Comment:
I really think this should be changed the way we discussed during the call without addcallback